### PR TITLE
Add support 3d ASTC compressed textures

### PR DESCRIPTION
**Pull request by lfedorov**
## Description

Some modern mobile games used 3d lightmaps in compressed formats, so right now RenderDoc can't collect them properly on GLES 2.0 / 3.0.
Added support NxNxN ASTC compressed textures (RGBA and SRGB8_ALPHA8). Small refactoring of existed code.

### CODE REVIEW

**Comment by lfedorov** [comment]
Here is sample Android project with 3d ASTC texture to test PL: 
[app-debug_compressed_3d_texture.zip](https://github.com/baldurk/renderdoc/files/7309937/app-debug_compressed_3d_texture.zip)

File: renderdoc/driver/gl/gl_resources.cpp [hunk 1]
```diff
@@ -262,31 +280,32 @@ size_t GetCompressedByteSize(GLsizei w, GLsizei h, GLsizei d, GLenum internalfor
...
-    return blocks[0] * blocks[1] * 16 * d;
+    return blocks[0] * blocks[1] * blocks[2] * 16;
   }
 
   RDCERR("Unrecognised compressed format %s", ToStr(internalformat).c_str());
   return GetByteSize(w, h, d, GetBaseFormat(internalformat), GetDataType(internalformat));
 }
 
-rdcpair<uint32_t, uint32_t> GetCompressedBlockSize(GLenum internalformat)
+rdcarray<uint32_t> GetCompressedBlockSize(GLenum internalformat)
```
Comment (baldurk) [inline renderdoc/driver/gl/gl_resources.cpp]: You should either use `rdcfixedarray` or else make a new struct with three members, as `rdcarray` will dynamically allocate its storage.

**Review by baldurk — CHANGES_REQUESTED** [review]
I think this looks good, I've only noted one case where you shouldn't use `rdcarray` to avoid unnecessary heap allocations.

File: renderdoc/driver/gl/gl_resources.cpp (hunk 1 above)
Comment (lfedorov) [reply renderdoc/driver/gl/gl_resources.cpp]: Sure! Fixed.

**Pull request merged**